vcd/resource_vcd_edgegateway.go

```diff
@@ -0,0 +1,201 @@
+package vcd
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+)
+
+func resourceVcdEdgeGateway() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdEdgeGatewayCreate,
+		Delete: resourceVcdEdgeGatewayDelete,
+		Read:   resourceVcdEdgeGatewayRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"org": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"vdc": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			/*
```

Comment:
Just not to forget to enable or delete with comment for future

Reply:
I left it here to get suggestions. Most likely I will delete these two properties. I saw that other entities have them in the API, but they are not exposed.

I have considered implementing "delete_force" manually, meaning that we remove links to other objects before deleting. But I am afraid that would be unsafe.